src/kernels/helpers.node.ts

```diff
@@ -1584,7 +1584,7 @@ export async function notifyAndRestartDeadKernel(
 export async function handleKernelError(
     serviceContainer: IServiceContainer,
     error: Error,
-    context: 'start' | 'interrupt' | 'restart' | 'execution',
+    errorContext: 'start' | 'interrupt' | 'restart' | 'execution',
```

Comment:
I renamed this, as there's already a global variable named `context` from mocha and that causes issues.
Basically i was using `context` in logging and it crashed during runtime, however ts never complained because that code path was passing the compiler due to it using the global `context` variable.

This way we dont run into such issues and TS compiler will still pick issues correctly

Reply:
Wow, good catch. Kinda scary that mocha was messing with type checking in the main product code. It's only for tests, right?

Reply:
>  It's only for tests, right?

Yes mocha is only used in tests, but its type def files declare a global variable that lets TS into thinking that those global vars are valid in the extension source as well. (I don't think its possible to work around, this unless issue)